codeblock in list isn't parsed correctly

~~~
error-tag:
:   access-denied

error-path:
:   Identifies the requested protocol operation.  The following example represents the
\<edit-config> protocol operation in the NETCONF base namespace:

    ```
    hallo
    ```
~~~
is not parsed correctly, because of the "`" it at least shows up as `<code>`.
When using `~~~` it even ignores that:

~~~
error-tag:
:   access-denied

error-path:
:   Identifies the requested protocol operation.  The following example represents the
\<edit-config> protocol operation in the NETCONF base namespace:

    ~~~
    hallo
    ~~~
~~~
AST:
~~~
Document ''
    List ''
        ListItem ''
            Paragraph ''
                Text 'error-tag:'
        ListItem ''
            Paragraph ''
                Text 'access-denied'
        ListItem ''
            Paragraph ''
                Text 'error-path:'
        ListItem ''
            Paragraph ''
                Text 'Identifies the requested protocol ope...'
                Text '<'
                Text 'edit-config> protocol operation in th...'
                Code '\nhallo\n'
                Text ''
~~~
That `Code` should be `CodeBlock`. My 1-line fix: `*flags |= ast.ListItemContainsBlock`, break this test:
~~~
--- FAIL: TestListWithMalformedFencedCodeBlock (0.00s)
    helpers_test.go:72: 
        Input   ["1. one\n\n    ```\n    code\n\n2. two\n"]
        Expected["<ol>\n<li>one\n```\ncode\n2. two</li>\n</ol>\n"]
        Actual  ["<ol>\n<li><p>one\n```\ncode\n2. two</p></li>\n</ol>\n"]
    helpers_test.go:72: 
        Input   ["1. one\n\n    ```\n    - code\n\n2. two\n"]
        Expected["<ol>\n<li>one\n```\n- code\n2. two</li>\n</ol>\n"]
        Actual  ["<ol>\n<li><p>one\n```\n- code\n2. two</p></li>\n</ol>\n"]
~~~

--- a/block_test.go
+++ b/block_test.go
@@ -1538,6 +1538,14 @@ func TestListWithMalformedFencedCodeBlock(t *testing.T) {
 	doTestsBlock(t, tests, parser.FencedCode)
 }
 
+func TestDefinitionListWithFencedCodeBlock(t *testing.T) {
+	var tests = []string{
+		"one:\n: def1\n\ntwo:\n: def2\n\n ~~~\ncode\n ~~~\n",
+		"<dl>\n<dt>one:</dt>\n<dd><p>def1</p></dd>\n<dt>two:</dt>\n<dd><p>def2</p>\n\n<pre><code>code\n</code></pre></dd>\n</dl>\n",
+	}
+	doTestsBlock(t, tests, parser.FencedCode|parser.DefinitionLists)
+}
+
 func TestListWithFencedCodeBlockNoExtensions(t *testing.T) {
 	// If there is a fenced code block in a list, and FencedCode is not set,
 	// lists should be processed normally.

--- a/parser/block.go
+++ b/parser/block.go
@@ -1627,7 +1627,10 @@ gatherlines:
 					// start of codeblock
 					codeBlockMarker = marker
 				} else {
-					// end of codeblock.
+					// end of codeblock, mark list as containing a block
+					if containsBlankLine {
+						*flags |= ast.ListItemContainsBlock
+					}
 					codeBlockMarker = ""
 				}
 			}

